Fails on darwin with:

```
                    except OSError as err:                                                                                                                                                  >                       raise OSError(err.errno, 'error while attempting '                                                                                                                                                        'to bind on address %r: %s'                                                                                                                                                                 % (sa, err.strerror.lower())) from None                                                                                                               E                                     PermissionError: [Errno 1] error while attempting to bind on address ('127.0.0.1', 42898): operation not permitted                                                                                                                                                                                                                                /nix/store/cwrsy4lr71pbs89krkfcwb3s4amc3rjg-python3-3.8.6/lib/python3.8/asyncio/base_events.py:1463: PermissionError                                                                        --------------------------- Captured stderr teardown --------------------------

....

electrum/tests/test_network.py::TestNetwork::test_can_connect_during_backward
electrum/tests/test_network.py::TestNetwork::test_chain_false_during_binary
electrum/tests/test_network.py::TestNetwork::test_fork_conflict
electrum/tests/test_network.py::TestNetwork::test_fork_noconflict
  /nix/store/cwrsy4lr71pbs89krkfcwb3s4amc3rjg-python3-3.8.6/lib/python3.8/asyncio/events.py:81: RuntimeWarning: coroutine 'ignore_exceptions.<locals>.wrapper' was never awaited
    self._context.run(self._callback, *self._args)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED electrum/tests/test_lntransport.py::TestLNTransport::test_loop - Permi...
============ 1 failed, 502 passed, 3 skipped, 4 warnings in 39.40s =============
--- Logging error ---
Traceback (most recent call last):
  File "/nix/store/cwrsy4lr71pbs89krkfcwb3s4amc3rjg-python3-3.8.6/lib/python3.8/logging/__init__.py", line 1084, in emit
    stream.write(msg + self.terminator)
ValueError: I/O operation on closed file.
Call stack:
  File "/nix/store/cwrsy4lr71pbs89krkfcwb3s4amc3rjg-python3-3.8.6/lib/python3.8/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/nix/store/cwrsy4lr71pbs89krkfcwb3s4amc3rjg-python3-3.8.6/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/private/tmp/nix-build-electrum-4.0.7.drv-0/Electrum-4.0.7/electrum/plugin.py", line 214, in run
    self.on_stop()
  File "/private/tmp/nix-build-electrum-4.0.7.drv-0/Electrum-4.0.7/electrum/util.py", line 357, in on_stop
    self.logger.info("stopped")
Message: 'stopped'
Arguments: ()
```

--- a/pkgs/applications/misc/electrum/default.nix
+++ b/pkgs/applications/misc/electrum/default.nix
@@ -19,15 +19,7 @@
 }:
 
 let
-  version = "4.0.6";
-
-  # electrum is not compatible with dnspython 2.0.0 yet
-  # use the latest 1.x release instead
-  py = python3.override {
-    packageOverrides = self: super: {
-      dnspython = super.dnspython_1;
-    };
-  };
+  version = "4.0.7";
 
   libsecp256k1_name =
     if stdenv.isLinux then "libsecp256k1.so.0"
@@ -43,7 +35,7 @@ let
     owner = "spesmilo";
     repo = "electrum";
     rev = version;
-    sha256 = "0mdbg2sq56nv0hx0rrcbgrv1lv89nqc6cqigivgk665hhjm4v5kq";
+    sha256 = "06vcbj9p96d8v4xjlygzr74lqllb9adn8k0racajzq61ijb0imi2";
 
     extraPostFetch = ''
       mv $out ./all
@@ -52,13 +44,13 @@ let
   };
 in
 
-py.pkgs.buildPythonApplication {
+python3.pkgs.buildPythonApplication {
   pname = "electrum";
   inherit version;
 
   src = fetchurl {
     url = "https://download.electrum.org/${version}/Electrum-${version}.tar.gz";
-    sha256 = "0sp8p720g3rqnh52ddhaw2v4hjgpxcwbackw9qc1g9xac1q0942d";
+    sha256 = "0k5xf97ga3ixd02g1y6v84hbxd8yhvpj5iz2rhxs8wfnkfwibzh4";
   };
 
   postUnpack = ''
@@ -68,19 +60,18 @@ py.pkgs.buildPythonApplication {
 
   nativeBuildInputs = stdenv.lib.optionals enableQt [ wrapQtAppsHook ];
 
-  propagatedBuildInputs = with py.pkgs; [
+  propagatedBuildInputs = with python3.pkgs; [
     aiohttp
     aiohttp-socks
     aiorpcx
     attrs
     bitstring
+    cryptography
     dnspython
-    ecdsa
     jsonrpclib-pelix
     matplotlib
     pbkdf2
     protobuf
-    pycryptodomex
     pysocks
     qrcode
     requests
@@ -121,10 +112,15 @@ py.pkgs.buildPythonApplication {
     wrapQtApp $out/bin/electrum
   '';
 
-  checkInputs = with py.pkgs; [ pytest ];
+  checkInputs = with python3.pkgs; [ pytestCheckHook pycryptodomex ];
 
-  checkPhase = ''
-    py.test electrum/tests
+  pytestFlagsArray = [ "electrum/tests" ];
+
+  disabledTests = [
+    "test_loop"  # test tries to bind 127.0.0.1 causing permission error
+  ];
+
+  postCheck = ''
     $out/bin/electrum help >/dev/null
   '';
 

